src/Avalonia.X11/X11CursorFactory.cs

```diff
@@ -67,12 +72,67 @@ public IPlatformHandle GetCursor(StandardCursorType cursorType)
             return new PlatformHandle(handle, "XCURSOR");
         }
 
+        public unsafe IPlatformHandle CreateCursor(IBitmapImpl cursor, PixelPoint hotSpot)
+        {
+            return new XImageCursor(_display, cursor, hotSpot);
+        }
+
         private static IntPtr GetNullCursor(IntPtr display)
         {
             XColor color = new XColor();
             IntPtr window = XLib.XRootWindow(display, 0);
             IntPtr pixmap = XLib.XCreateBitmapFromData(display, window, NullCursorData, 1, 1);
             return XLib.XCreatePixmapCursor(display, pixmap, pixmap, ref color, ref color, 0, 0);
         }
+
+        private unsafe class XImageCursor : IFramebufferPlatformSurface, IPlatformHandle, IDisposable
```

Comment:
I'd suggest to create a standard `LockedFramebuffer` wrapper instead of implementing `IFramebufferPlatformSurface`. That way we won't need to retain the `_blob` in a field and dispose said wrapper instead.